docs/developer/goal.md

```diff
@@ -0,0 +1,104 @@
+
+```
+    const BaseGoals = goals("checks")
+        .plan(new AutoCodeInspection())
+        .plan(AutofixGoal);
+```
+
+You can specify ordering, if some goals should wait for others to succeed. Here, we don't want to start the build until after Autofixes have completed.
+If the autofixes do anything, they'll make a new commit, and we don't bother building this one.
+
+```
+    const BuildGoals = goals("build")
+        .plan(new Build().with({ name: "Maven", builder: new MavenBuilder(sdm) }))
+        .after(AutofixGoal);
+```
+
+Finally, you can tell the SDM which goal sets to run on . Here, we set the BaseGoals (inspection and autofix) on every push. Then if 
+this is a Maven project (identified by having a pom.xml), we do the build as well.
+
+```
+    sdm.addGoalContributions(goalContributors(
+        onAnyPush().setGoals(BaseGoals),
+        whenPushSatisfies(IsMaven).setGoals(BuildGoals),
+    ));
+```
+
+Check the page about [Setting Goals][setting-goals] for more details.
+
+[setting-goals]: set-goals.md (Setting Goals in an SDM)
+
+## Built-in Goals
+
+A goal object has some identifying information, code to fulfill the goal, and sometimes preconditions (goals that need to complete before this one can go). Some common ones have their own constructors:
+
+### AutoInspect
+
+Run an inspection on the code; if the code doesn't pass, you can fail the goals or require approval (a button push). To use it, you'll need to create one, set it on each push, and register inspections on it.
+
+Instantiate an empty one:
+
+```typescript
+export const CodeInspectionGoal = new AutoCodeInspection();
+```
+
+And set it when you want it to run on a push. Here's the shortest way to run this goal on every push:
+
+```typescript
+    sdm.addGoalContributions(goalContributors(
+        onAnyPush().setGoals(goals("Inspections").plan(CodeInspectionGoal))))
+```
+
+Now the fun part: register inspections on it. Check the [Inspections][inspection] page for more on how to write inspections.
+Once you have an [AutoInspectRegistration][AutoInspectRegistration], register it on your goal:
+
+```typescript
+CodeInspectionGoal.with(MyAutoInspectRegistration)
+    .with(AnotherInspectRegistration);
+```
+
+You can register any number of inspections. You can call `with` on the goal at any point in SDM configuration.
+
+If no inspections are registered, the goal will succeed. If any registration's `onInspectionResult` returns "fail", the goal will fail. If none return "fail" but one returns "require approval", the goal will go to Waiting for Approval state until someone clicks the Approve button in Slack or on the Atomist dashboard. 
+
+[AutoInspectRegistration]: https://atomist.github.io/sdm/interfaces/_lib_api_registration_autoinspectregistration_.autoinspectregistration.html (AutoInspectRegistration API Doc)
+[inspection]: inspect.md (Automatic Code Inspections)
+
+### Autofix
```

Comment:
It seems these all also have empty pages associated with them too. Are they needed here?

Reply:
you're right, a bulleted list with links to those pages would be better

Reply:
oh ... actually they are needed here, and as anchors. AutoInspect is populated already, and says how to instantiate and use the goal class.